lightning/src/ln/channelmanager.rs

```diff
@@ -8154,132 +8225,23 @@ mod tests {
-			temporary_channel_id: open_channel_msg.temporary_channel_id,
-			funding_txid: Txid::from_hex("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff").unwrap(),
-			funding_output_index: 0,
-			signature: signature,
-		};
-
-		let funding_signed_msg = msgs::FundingSigned {
-			channel_id: channel_id,
-			signature: signature,
-		};
-
-		let channel_ready_msg = msgs::ChannelReady {
-			channel_id: channel_id,
-			next_per_commitment_point: unkown_public_key,
-			short_channel_id_alias: None,
-		};
-
-		let announcement_signatures_msg = msgs::AnnouncementSignatures {
-			channel_id: channel_id,
-			short_channel_id: 0,
-			node_signature: signature,
-			bitcoin_signature: signature,
-		};
-
-		let channel_reestablish_msg = msgs::ChannelReestablish {
-			channel_id: channel_id,
-			next_local_commitment_number: 0,
-			next_remote_commitment_number: 0,
-			data_loss_protect: OptionalField::Absent,
-		};
-
-		let closing_signed_msg = msgs::ClosingSigned {
-			channel_id: channel_id,
-			fee_satoshis: 1000,
-			signature: signature,
-			fee_range: None,
-		};
-
-		let shutdown_msg = msgs::Shutdown {
-			channel_id: channel_id,
-			scriptpubkey: Script::new(),
-		};
-
-		let onion_routing_packet = msgs::OnionPacket {
-			version: 255,
-			public_key: Ok(unkown_public_key),
-			hop_data: [1; 20*65],
-			hmac: [2; 32]
-		};
-
-		let update_add_htlc_msg = msgs::UpdateAddHTLC {
-			channel_id: channel_id,
-			htlc_id: 0,
-			amount_msat: 1000000,
-			payment_hash: PaymentHash([1; 32]),
-			cltv_expiry: 821716,
-			onion_routing_packet
-		};
-
-		let commitment_signed_msg = msgs::CommitmentSigned {
-			channel_id: channel_id,
-			signature: signature,
-			htlc_signatures: Vec::new(),
-		};
-
-		let update_fee_msg = msgs::UpdateFee {
-			channel_id: channel_id,
-			feerate_per_kw: 1000,
-		};
-
-		let malformed_update_msg = msgs::UpdateFailMalformedHTLC{
-			channel_id: channel_id,
-			htlc_id: 0,
-			sha256_of_onion: [1; 32],
-			failure_code: 0x8000,
-		};
-
-		let fulfill_update_msg = msgs::UpdateFulfillHTLC{
-			channel_id: channel_id,
-			htlc_id: 0,
-			payment_preimage: PaymentPreimage([1; 32]),
-		};
-
-		let fail_update_msg = msgs::UpdateFailHTLC{
-			channel_id: channel_id,
-			htlc_id: 0,
-			reason: msgs::OnionErrorPacket { data: Vec::new()},
-		};
-
-		let revoke_and_ack_msg = msgs::RevokeAndACK {
-			channel_id: channel_id,
-			per_commitment_secret: [1; 32],
-			next_per_commitment_point: unkown_public_key,
-		};
-
 		// Test the API functions and message handlers.
 		check_not_connected_to_peer_error(nodes[0].node.create_channel(unkown_public_key, 1_000_000, 500_000_000, 42, None), unkown_public_key);
 
-		nodes[1].node.handle_open_channel(&unkown_public_key, &open_channel_msg);
+		check_unkown_peer_error(nodes[0].node.accept_inbound_channel(&channel_id, &unkown_public_key, 42), unkown_public_key);
 
-		nodes[0].node.handle_accept_channel(&unkown_public_key, &accept_channel_msg);
```

Comment:
Ideally I'd like to keep these msgs handling function calls with an unknown peer. Looking into if there's a good way of expecting a panic with a given message here, without having to do a separate `test` for every msg handling function call.